lib/src/dart2_9_suggestors/dart2_9_utilities.dart

```diff
@@ -0,0 +1,164 @@
+  if (method is MethodInvocation && method.methodName.name == 'connect') {
+    final generatedArg = args.first;
+    if (generatedArg is SimpleIdentifier) {
+      return generatedArg.name.startsWith(generatedPrefix)
+          ? generatedArg
+          : null;
+    } else if (generatedArg is MethodInvocation &&
+        generatedArg.methodName.name == castFunctionName) {
+      final arg = generatedArg.argumentList.arguments.first;
+      return arg is SimpleIdentifier && arg.name.startsWith(generatedPrefix)
+          ? arg
+          : null;
+    }
+  }
+
+  return null;
+}
+
+/// Returns the generated factory config argument from [argList].
+///
+/// Used specifically to find the generated factory config in `uiFunction` /
+/// `uiForwardRef` usages.
+///
+/// Returns `null` if no generated argument found.
+///
+/// Example:
+///
+/// | input                                                  | output      |
+/// |--------------------------------------------------------|-------------|
+/// | ((props) {}, $FooConfig)                               | $FooConfig  |
+/// | ((props) {}, _$FooConfig)                              | _$FooConfig |
+/// | ((props) {}, UiFactoryConfig())                        | null        |
+SimpleIdentifier getGeneratedFactoryConfigArg(ArgumentList argList) {
+  final args = argList.arguments;
+  if (args.length != 2) return null;
+
+  final configPattern = RegExp(r'_?\$\S*Config$');
+
+  final generatedArg = args[1];
+  return generatedArg is SimpleIdentifier &&
+          generatedArg.name.startsWith(configPattern)
+      ? generatedArg
+      : null;
+}
+
+/// Returns whether or not [node] is a class component factory declaration.
+bool isClassComponentFactory(TopLevelVariableDeclaration node) {
+  final type = node.variables?.type;
+  if (type != null && type is NamedType && type?.name?.name == 'UiFactory') {
+    final initializer = node.variables?.variables?.first?.initializer;
+    if (initializer is SimpleIdentifier) {
+      return initializer.name.startsWith('_\$');
+    } else if (initializer is MethodInvocation &&
+        initializer.methodName.name == castFunctionName) {
+      return initializer.argumentList.arguments.first
+          .toSource()
+          .startsWith('_\$');
+    }
+  }
+
+  return false;
+}
+
+/// Returns whether or not [node] is in the legacy boilerplate syntax.
+bool isLegacyFactoryDecl(TopLevelVariableDeclaration node) {
+  final annotation = node.metadata?.firstWhere(
+      (m) => m.toSource().startsWith('@Factory'),
+      orElse: () => null);
+  return isClassComponentFactory(node) && annotation != null;
+}
+
+/// Iterates through the [comment] token stream and removes all ignore
+/// comments containing [ignoreToRemove] using the [yieldPatch] provided.
+///
+/// The entire ignore comment will only be removed if [ignoreToRemove] is the only
+/// item in the list to ignore. Otherwise, [ignoreToRemove] will just be
+/// removed from the list.
+///
+/// Example:
+///
+/// Calling:
```

Comment:
~Are there meant to be examples between `Example` and `Calling`?~

Ohh I get it, nevermind. #nit I think the wording could be clarified or could use indentation, but I caught up! 😄